### PR TITLE
*: fix golangci-lint errors

### PR DESCRIPTION
**Pull request by AlexNPavel**
Fix all linting errors detected by golangci-lint v1.61.0.

### CODE REVIEW

**Comment by AlexNPavel** [comment]
I've added anther commit running `goimports` on all non-vendor files to clean up the import blocks

File: cmd/release-controller/sync_release.go [hunk 1]
```diff
@@ -370,42 +370,3 @@ func jobIsComplete(job *batchv1.Job) (succeeded bool, complete bool) {
...
-
-		job.Annotations[releasecontroller.ReleaseAnnotationSource] = mirror.Annotations[releasecontroller.ReleaseAnnotationSource]
-		job.Annotations[releasecontroller.ReleaseAnnotationTarget] = mirror.Annotations[releasecontroller.ReleaseAnnotationTarget]
-		job.Annotations[releasecontroller.ReleaseAnnotationGeneration] = strconv.FormatInt(release.Target.Generation, 10)
-		job.Annotations[releasecontroller.ReleaseAnnotationReleaseTag] = mirror.Annotations[releasecontroller.ReleaseAnnotationReleaseTag]
-
-		klog.V(2).Infof("Running release mirror job %s/%s for %s to %s", c.jobNamespace, job.Name, name, toImage)
-		return job, nil
-	})
-}
```
Comment (bradmwilliams) [inline cmd/release-controller/sync_release.go:411]: This function was introduced as the first part (https://github.com/openshift/release-controller/pull/630) of an ongoing effort that I am working on.  Can you leave this in place?

**Review by bradmwilliams — CHANGES_REQUESTED** [review]

File: cmd/release-controller/sync_release.go (hunk 1 above)
Comment (bradmwilliams) [reply cmd/release-controller/sync_release.go:411]: Is there a way to tell golangci-lint to ignore this call (for now)?  That way we can get https://github.com/openshift/release/pull/58528 operational?  If not, then I suppose I can just pull the function back into my other PR.

Comment (AlexNPavel) [reply cmd/release-controller/sync_release.go]: Yes, I've readded the function with a `// nolint:unused` comment to have golangci-lint ignore it for now.

Comment (bradmwilliams) [reply cmd/release-controller/sync_release.go]: Excellent!  Thanks

**Review by bradmwilliams — APPROVED** [review]

**Pull request merged**